modules/virt-binding-devices-vfio-driver.adoc

```diff
@@ -0,0 +1,132 @@
+<5> Description needed. What does the value 420 for `mode` do for this source file? Please provide details.
+<6> Description needed. When `mode` is set to `true`, is there a default file that is overwritten by the `/etc/modprobe.d/vfio.conf` file? If that's not correct, please provide details.
+<7> The path for the `vfio.conf` file.
+<8> The encoded text for `source` (`dmZpby1wY2kK`) that translates to `options vfio-pci" ids=10de:1eb8|base64`.
+<9> The file with the `vfio-pci` configuration that loads the PCI device with the VFIO driver.
+
+. Create a `MachineConfig` object file that identifies the kernel argument for the IOMMU configuration:
++
+[source,terminal]
+----
+$ oc create -f 100-worker-kernel-arg-iommu.yaml
+----
+
+. Verify that the `MachineConfig` object file was added.
++
+[source,terminal]
+----
+$ oc get MachineConfig
+----
++
+.Example output
+[source, terminal]
+----
+NAME                             GENERATEDBYCONTROLLER                      IGNITIONVERSION  AGE
+00-master                        d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+00-worker                        d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+01-master-container-runtime      d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+01-master-kubelet                d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+01-worker-container-runtime      d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+01-worker-kubelet                d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+100-worker-iommu                                                            3.2.0            30s
+100-worker-vfiopci-configuration                                            3.2.0            30s
+
+----
+
+. Use the `lspci` command to obtain the `vendor-ID` and the `device-ID` for the PCI device.
++
+[source, terminal]
+----
+$ lspci -nnv | grep -i nvidia
+----
+.Example output
+[source, terminal]
+02:01.0 3D controller [0302]: NVIDIA Corporation GV100GL [Tesla V100 PCIe 32GB] [10de:1eb8] (rev a1)
+
+. Edit the VFIO configuration in the `etc/modprobe.d/vfio.conf` file by adding the values for the `vendor-ID` and the `device-ID`.
++
+[source,terminal]
+----
+$ echo "options vfio-pci ids=10de:1eb8 > /etc/modprobe.d/vfio/conf"
+----
+These values are converted into the `base64 format` and encoded.
++
+.Example output
+[source,terminal]
+----
+$ data:text/plain;charset=utf-8;base64,b3B0aW9ucyB2ZmlvLXBjaSBpZHM9MTBkZToxZWI4Cg==
+----
+
+. Update the VFIO configuration file to load the VFIO driver after it is rebooted.
++
+.Example output
+[source,terminal]
+----
+$ echo 'vfio-pci' > /etc/modules-load.d/vfio-pci.conf
```

Comment:
here too, we don't need to edit this file.
Instead, we need to encode `vfio-pci` and add this to the `source` in the `MachineConfig` object  <8>

`$echo vfio-pci|base64`
the output will be `dmZpby1wY2kK`

data:text/plain;charset=utf-8;base64,dmZpby1wY2kK should be added to <8>

Reply:
Updated the content.